src/components/Section/SubstanceUse/SubstanceUseConnector.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import {
+  updateApplication,
+  reportErrors,
+} from 'actions/ApplicationActions'
+
+const connectSubstanceUseSection = (Component, {
+  section, subsection, store, storeKey,
+}) => {
+  class ConnectedSubstanceUseSection extends React.Component {
+    constructor(props) {
+      super(props)
+
+      this.section = section
+      this.subsection = subsection
+      this.store = store
+    }
+
+    handleError = (value, arr) => {
+      const { dispatch } = this.props
+      const action = reportErrors(this.section, this.subsection, arr)
+      dispatch(action)
+      return arr
+    }
+
+    handleUpdate = (field, values) => {
+      const { dispatch } = this.props
+      dispatch(updateApplication(this.store, field, values))
+    }
+
+    render() {
+      return (
+        <Component
+          onUpdate={this.handleUpdate}
+          onError={this.handleError}
+          {...this.props}
+        />
+      )
+    }
+  }
+
+  ConnectedSubstanceUseSection.propTypes = {
+    dispatch: PropTypes.func, // Passed in via connect (below)
+  }
+
+  ConnectedSubstanceUseSection.defaultProps = {
+    dispatch: () => {},
+  }
+
+  const mapStateToProps = (state) => {
+    const app = state.application || {}
+    const substance = app.Substance || {}
+    const errors = app.Errors || {}
+    const completed = app.Completed || {}
+
+    switch (storeKey) {
```

Comment:
It looks like the `addressBooks` prop should also be passed to `OrderedTreatments, VoluntaryTreatments, OrderedCounselings, VoluntaryCounselings`